src/Backoffice/Tasks/Domain/Actions/UpdateTaskAction copy.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lightit\Backoffice\Tasks\Domain\Actions;
+
+use Lightit\Backoffice\Tasks\App\Notifications\TaskAssignmentNotification;
+use Lightit\Backoffice\Tasks\Domain\DataTransferObjects\TaskDto;
+use Lightit\Backoffice\Tasks\Domain\Models\Task;
+
+class UpdateTaskAction
+{
+    public function execute(Task $task, TaskDto $taskDto): Task
+    {
+        $task = Task::find($task);
+
+        if (! $task) {
+            throw new \Exception('Task not found.');
+        }
```

Comment:
I think here we can avoid the exception by doing this

```suggestion
        $task = Task::findOrFail($task);
```